### ISSUE TITLE
Type of `tuple` is partially unknown

### ISSUE THREAD

**Issue by Molkree** [microsoft/pylance-release#2140]
## Environment data

-   Language Server version: v2021.12.0
-   OS and version: Win10
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

## Expected behaviour

No errors.

## Actual behaviour

![image](https://user-images.githubusercontent.com/12860284/144714027-66cbab4e-d1ff-40c0-8e09-3d0d3000f2ec.png)

## Code Snippet / Additional information

```python
nums: list[int] = [1, 2, 3]
max_product, min_product = nums[0], nums[0]
for x in nums[1:]:
    candidates = x, max_product * x, min_product * x
    min_product = min(candidates)
    max_product = max(candidates)
```
This behavior wasn't present in the v2021.11.2 version.

**Comment by erictraut** [comment]
Thanks for reporting the problem. This will be fixed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2021.12.1, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2021121-9-december-2021)
